### ISSUE TITLE
[Bug] [kompile] - gen-bison-parser and --bison-lists cause parse error when running

### ISSUE THREAD

**Issue by hjorthjort** [runtimeverification/k#1609]
K Version
---------

```
RV-K version 1.0-SNAPSHOT
Git revision: 0aa62f7
Git branch: UNKNOWN
Build date: Tue Oct 13 12:24:05 UTC 2020
```

Description
-----------

Parser generated with  `--bison-lists` get IndexOutOfBoundsException when run

Input Files
-----------

File: `main.md`

```k
module MAIN-SYNTAX
 
    syntax Stmts      ::= NeList{Stmt      , ""} [klabel(listStmt)]
    syntax Stmt       ::= ModuleDecl
    syntax ModuleDecl ::= "(" "module" ")"
    
    
endmodule

module MAIN
    imports MAIN-SYNTAX
    configuration
        <main> $PGM:Stmts </main>
endmodule
```

File: `test.wat`

```
( module )
( module )
```

Reproduction Steps
------------------

```
$ kompile --backend llvm --md-selector k --gen-bison-parser --bison-lists --debug main.md --directory .build/defn/llvm --main-module MAIN --syntax-module MAIN-SYNTAX
```

Running on LLVM backend cause error

```
$ krun ../test.wat --directory .build/defn/llvm --debug
Stdout:

Stderr:
[Error] Internal: Uncaught exception thrown of type IndexOutOfBoundsException.
Please rerun your program with the --debug flag to generate a stack trace, and
file a bug report at https://github.com/kframework/k/issues
(IndexOutOfBoundsException: null)

        at org.kframework.utils.errorsystem.KEMException.create(KEMException.java:130)
        at org.kframework.utils.errorsystem.KEMException.criticalError(KEMException.java:39)
        at org.kframework.krun.KRun.externalParse(KRun.java:238)
        at org.kframework.krun.KRun.parseConfigVars(KRun.java:162)
        at org.kframework.krun.KRun.run(KRun.java:78)
        at org.kframework.krun.KRunFrontEnd.run(KRunFrontEnd.java:95)
        at org.kframework.main.FrontEnd.main(FrontEnd.java:62)
        at org.kframework.main.Main.runApplication(Main.java:119)
        at org.kframework.main.Main.runApplication(Main.java:109)
        at org.kframework.main.Main.main(Main.java:57)
```

Expected Behavior
-----------------
Should parse correctly. Without `--bison-lists`, it does, producing this:
```
<main>
  ( module )  ( module )  .Stmts
</main>
```

**Comment by hjorthjort** [comment]
Reproduced with newer version:
```
RV-K version 1.0-SNAPSHOT
Git revision: adb796d
Git branch: master
Build date: Wed Oct 21 12:35:30 UTC 2020
```

**Comment by ehildenb** [comment]
Perhaps the code which parses lists as lassoc, and then corrects it to rassoc lists, is not being done correctly when calling `kast` on grammars generated with `--bison-lists` production? It's true that this should work, but it's a strange use-case.

@hjorthjort it's confusing why we are (i) generating a bison parser, and then (ii) not using it by calling `krun` instead of `kx` or `llvm-krun`. Can you clarify if that's intentional?

**Comment by ehildenb** [comment]
This appears to be fixed on K 5.4.30.

File `test.k`:

```
module TEST-SYNTAX

    syntax Stmts      ::= NeList{Stmt      , ""} [klabel(listStmt)]
    syntax Stmt       ::= ModuleDecl
    syntax ModuleDecl ::= "(" "module" ")"


endmodule
 
module TEST
    imports TEST-SYNTAX
    configuration
        <main> $PGM:Stmts </main>
endmodule
```

File `test.wat`:

```
( module )
( module )
```

Then I get the following interaction:

```
% kompile --backend llvm --gen-bison-parser --bison-lists test.k
 
% krun test.wat
<main>
  .Stmts  ( module )  ( module )
</main>
```

Closing as fixed.
